### ISSUE TITLE
F.grid_sample produces weird results on single-pixel images

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#42625]
## 🐛 Bug

```
import torch
import torch.nn.functional as F

B = 1
src_image = torch.ones([B, 1, 1, 1])
grid = -5 * torch.ones([B, 3, 5, 2])
img = F.grid_sample(src_image, grid, align_corners=True, padding_mode="zeros")
print(img)
```
The above returns an image full of 1's. However, some would expect it to return an image full of zeros -- each grid coordinate is (-5, -5), which is definitely beyond the [-1, 1] boundaries for the image, so the value should be the padding value.

I'm not completely convinced that this is a bug. When `align_corners=True` and the image only has a single pixel, each of the corners (-1, -1), (1, -1), (-1, 1), (1, 1) are mapped to the center of the pixel. Since they're all the same location, the geometry becomes weird -- if the distance between (-1, -1) and (1, 1) is 0, then what's the distance between (-5, -5) and (-1, -1)?

Perhaps the way to resolve this is to add a warning in the documentation and/or add a warning to the code if any inputs have height or width 1.

## Environment

PyTorch master

**Comment by zou3519** [comment]
If we just go by the documentation, "If grid has values outside the range of [-1, 1], the corresponding outputs are handled as defined by padding_mode.", then the above is silently incorrect behavior.

**Comment by zou3519** [comment]
tagging some potentially interested folks from some past grid_sample threads: @fmassa, @ailzhang, @SsnL
